src/placement/ring_map.c

```diff
@@ -487,6 +487,9 @@ ring_create(struct pl_ring_map *rimap, unsigned int index,
 		return -DER_NOMEM;
 
 	first = pool_map_targets(rimap->rmp_map.pl_poolmap);
+	if (first == NULL)
+		return DER_INVAL;
```

Comment:
would NOMEM be better?

Reply:
pool_map_targets() internally makes a call to:
`pool_map_find_target(map,` PO_COMP_ID_ALL, &targets);`
which will return either a pointer to the start of the pools target array, or null if there weren't any targets.

I was using INVAL because we can't compute placement if the pool doesn't contain any targets.
Would NOMEM be more appropriate in this case?

Reply:
@jolivier23

Reply:
It seems impossible to return NULL here. why not D_ASSERT here? or I miss sth?

Reply:
It will only return NULL when there are no targets in the pool map.
is there any case where the pool map wouldn't contain targets that would shouldn't call an assert?